server/service/src/program/query.rs

```diff
@@ -38,3 +39,38 @@ pub fn get_program(
         Err(SingleRecordError::NotFound(id))
     }
 }
+
+pub fn get_periods(
+    connection: &StorageConnection,
+    store_id: String,
+    program_id: Option<String>,
+) -> Result<ListResult<PeriodRow>, ListError> {
+    let periods = if let Some(program_id) = program_id {
+        let period_schedule_ids = ProgramRequisitionSettingsRowRepository::new(connection)
+            .find_many_by_program_id(&program_id)?
+            .iter()
+            .map(|settings| settings.period_schedule_id.clone())
+            .collect::<Vec<String>>();
+
+        PeriodRepository::new(connection)
+            .query_by_filter(
+                store_id,
+                None,
+                PeriodFilter::new().period_schedule_id(EqualFilter::equal_any(period_schedule_ids)),
+            )?
+            .iter()
+            .map(|period| period.period_row.clone())
+            .collect::<Vec<PeriodRow>>()
+    } else {
+        PeriodRepository::new(connection)
+            .query_by_filter(store_id, None, PeriodFilter::new())?
+            .iter()
+            .map(|period| period.period_row.clone())
```

Comment:
Just wondering if this would give UI issues if we are looking at say weekly periods across a year without any pagination?

Reply:
There's a comment about this in other PR about creating issue for pagination. Will do when I get back